codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeadersFactory.java

```diff
@@ -0,0 +1,327 @@
+                            || HttpHeaderNames.TRANSFER_ENCODING.contentEqualsIgnoreCase(name)
+                            || HttpHeaderNames.TRAILER.contentEqualsIgnoreCase(name)) {
+                        throw new IllegalArgumentException("prohibited trailing header: " + name);
+                    }
+                }
+            };
+
+    @SuppressWarnings("unchecked")
+    private static final NameValidator<CharSequence> NO_NAME_VALIDATOR = NameValidator.NOT_NULL;
+    @SuppressWarnings("unchecked")
+    private static final ValueValidator<CharSequence> NO_VALUE_VALIDATOR =
+            (ValueValidator<CharSequence>) ValueValidator.NO_VALIDATION;
+
+    private static final DefaultHttpHeadersFactory DEFAULT = new DefaultHttpHeadersFactory();
+    private static final DefaultHttpHeadersFactory DEFAULT_TRAILER =
+            new DefaultHttpHeadersFactory(DEFAULT_TRAILER_NAME_VALIDATOR, DEFAULT_VALUE_VALIDATOR, false);
+    private static final DefaultHttpHeadersFactory DEFAULT_COMBINING =
+            new DefaultHttpHeadersFactory(DEFAULT.nameValidator, DEFAULT.valueValidator, true);
+    private static final DefaultHttpHeadersFactory DEFAULT_NO_VALIDATION =
+            new DefaultHttpHeadersFactory(NO_NAME_VALIDATOR, NO_VALUE_VALIDATOR, false);
+
+    private final NameValidator<CharSequence> nameValidator;
+    private final ValueValidator<CharSequence> valueValidator;
+    private final boolean combiningHeaders;
+
+    /**
+     * Create a header builder with the default settings.
+     */
+    private DefaultHttpHeadersFactory() {
+        this(DEFAULT_NAME_VALIDATOR, DEFAULT_VALUE_VALIDATOR, false);
+    }
+
+    /**
+     * Create a header builder with the given settings.
+     *
+     * @param nameValidator The name validator to use, not null.
+     * @param valueValidator The value validator to use, not null.
+     * @param combiningHeaders {@code true} if multi-valued headers should be combined into single lines.
+     */
+    private DefaultHttpHeadersFactory(
+            NameValidator<CharSequence> nameValidator,
+            ValueValidator<CharSequence> valueValidator,
+            boolean combiningHeaders) {
+        this.nameValidator = checkNotNull(nameValidator, "nameValidator");
+        this.valueValidator = checkNotNull(valueValidator, "valueValidator");
+        this.combiningHeaders = combiningHeaders;
+    }
+
+    /**
+     * Create a new header builder instance with the given overrides.
+     * This method is used by all other {@code with-} methods, for creating new builder instances.
+     *
+     * @param nameValidator The name validator to use, not null.
+     * @param valueValidator The value validator to use, not null.
+     * @param combiningHeaders {@code true} if multi-valued headers should be combined into single lines.
+     * @return The new header builder instance.
+     */
+    private static DefaultHttpHeadersFactory with(
```

Comment:
It was useful in the previous revision as a public method, but now it's private and identical to a ctor. Consider removing it in a favor of using a ctor